Make gempak a metatask

WalterKolczynski-NOAA opened the pull request:
# Description
Converts the regular gempak job into a metatask that runs one forecast hour at a time. The forecast hour to process is exported as a zero-padded string from rocoto as `FHR3`.

This assumes gempak is needed at all forecast output times. If gempak needs fewer times, additional changes will be needed.

Refs NOAA-EMC/global-workflow#1250
In contribution to NOAA-EMC/global-workflow#2671

# Type of change
- Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?
- Workflow generation only

# Checklist
- [ ] Any dependent changes have been merged and published
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] I have made corresponding changes to the documentation if necessary

WalterKolczynski-NOAA committed:
Make gempak a metatask

Converts the regular gempak job into a metatask that runs one
forecast hour at a time. The forecast hour to process is exported
from rocoto as `FHR3`.

This assumes gempak is needed at all forecast output times. If
gempak needs fewer times, additional changes will be needed.

Refs: NOAA-EMC/global-workflow#1250

WalterKolczynski-NOAA commented:
Note: this only does the main `gempak` job. If the follow-on jobs like `gempakmeta` also need the same treatment, I can help with that too.

GwenChen-NOAA commented on workflow/rocoto/gfs_tasks.py:
```diff
@@ -1461,13 +1466,20 @@ def gempak(self):
     def gempakmeta(self):
         deps = []
-        dep_dict = {'type': 'task', 'name': f'{self.cdump}gempak'}
+        dep_dict = {'type': 'metatask', 'name': f'{self.cdump}gempak'}
```
@WalterKolczynski-NOAA, does this change the gempakmeta job to a metatask and run one forecast hour per task too?  The gempakmeta job runs less than 15 minutes, so I didn't make any changes.  It needs to process all forecast hours at once because gempak metafiles contain multiple lead times.

WalterKolczynski-NOAA replied:
No. This is the dependency for the `gempakmeta` job; `gempak` is a metatask now, so we had to change the type of all the jobs that depend on it.

GwenChen-NOAA commented on workflow/rocoto/gfs_tasks.py:
```diff
@@ -1490,7 +1502,7 @@ def gempakmeta(self):
     def gempakmetancdc(self):
         deps = []
-        dep_dict = {'type': 'task', 'name': f'{self.cdump}gempak'}
+        dep_dict = {'type': 'metatask', 'name': f'{self.cdump}gempak'}
```
Same comment as for the gempakmeta job, no change was made for the gempakmetancdc job.

WalterKolczynski-NOAA replied:
Same response. These are the dependencies on the `gempak` job that _was_ changed.

GwenChen-NOAA commented on workflow/rocoto/gfs_tasks.py:
```diff
@@ -1513,7 +1525,7 @@ def gempakmetancdc(self):
     def gempakncdcupapgif(self):
         deps = []
-        dep_dict = {'type': 'task', 'name': f'{self.cdump}gempak'}
+        dep_dict = {'type': 'metatask', 'name': f'{self.cdump}gempak'}
```
Same comment as for the gempakmeta job, no change was made for the gempakncdcupapgif job.

GwenChen-NOAA commented on workflow/rocoto/gfs_tasks.py:
The gempakpgrb2spec job has updated to process one forecast hour per task in global-workflow [PR #2671](https://github.com/NOAA-EMC/global-workflow/pull/2671). Does its `dep_dict` need to be changed?

WalterKolczynski-NOAA replied:
If you want to change it to a metatask, it's not the `dep_dict` that would need to be changed, it is the `task_dict`. I can add that to this, or you can follow what I did for `gempak` as a guide.

GwenChen-NOAA replied:
This job is downstream from the npoess job, and it currently processes all forecast hours at once. If we change the gempakpgrb2spec job to run one forecast hour per task, does the npoess job need to be changed too?

WalterKolczynski-NOAA replied:
It doesn't have to be, but then you'd still be waiting for the npoess job to finish completely before starting the gempak one, so it'd probably be a good idea. But it also doesn't all have to be done at once in the same PR.

GwenChen-NOAA replied:
Ok, I will look at the code to see if it can be easily done. The npoess job does two things: 1) generate npoess files up to f024 and 2) generate goessimpgrb2 files from goesmaster files up to f180. So, it's not as straightforward as the gempak jobs. Can you add the change for the gempakpgrb2spec job into this PR?

WalterKolczynski-NOAA replied:
Yes. Should only take a minute.

We can also change the dependency to be a file instead of a job. The trick is just to make sure the file is complete before we start using it.

GwenChen-NOAA replied:
Thanks much!

WalterKolczynski-NOAA replied:
> Ok, I will look at the code to see if it can be easily done. The npoess job does two things: 1) generate npoess files up to f024 and 2) generate goessimpgrb2 files from goesmaster files up to f180. So, it's not as straightforward as the gempak jobs. Can you add the change for the gempakpgrb2spec job into this PR?

Sounds like the npoess job should be two separate jobs. We can tackle that later.

I'm adding a new `FHMAX_GOES` to control the bounds for the `gempakpgrb2spec` job so it isn't just hard-coded in the npoess job and we can use it here.

WalterKolczynski-NOAA replied:
Should be updated now. I added new variables to control when the GOES output ends and how frequent it is so we could use them.

GwenChen-NOAA replied:
Thanks, @WalterKolczynski-NOAA! Your changes look good to me. I will keep the npoess job as is.

WalterKolczynski-NOAA committed:
Turn gempack special job into a metatask

Converts the `gempakpgrb2spec` job into a metatask. To facilitate
this, new variables are created to control the length and cadence
of the GOES output.
